### PR TITLE
System 2449/fix gamepad

### PR DESCRIPTION
**Pull request by cmajed**
## Description
Rework Gamepad and Gamepad Management to fix:
        - handle gamepad pugged/unplugged
        - catch and handle confirmation
        - send correct vinput gamepad redis message
        - Handle axis changes
        - Rework catch and dispatch event to dispatch it only when axes changed
        - Add vendor id and product id to the plugg in command
        
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I've read & comply with the [contributing guidelines](https://github.com/Genymobile/genymotion-device-web-player/blob/main/CONTRIBUTING.md)
- [ ] I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] I have made corresponding changes to the documentation (README.md).
- [ ] I've checked my modifications for any breaking changes.

### CODE REVIEW

**Review by jparez — APPROVED** [review]
LGTM

**Comment by pgivel** [comment]
I think the first commit should not be in this PR ? Or maybe this PR should not target `main` ?

File: src/plugins/GamepadManager.js [hunk 1]
```diff
@@ -257,29 +257,23 @@ module.exports = class GamepadManager {
     }
 
     /**
-     * Handler for gamepad connection. Reemits a custom event with the parsed gamepad.
+     * Handler for gamepad connection/disconnection. Reemits a custom event with the parsed gamepad.
      * User class must then call listenForInputs in order to provide the remote index & start listening for inputs.
      * @param {GamepadEvent} event raw event coming from the browser Gamepad API
      */
     onGamepadConnected(event) {
```
Comment (pgivel) [inline src/plugins/GamepadManager.js:264]: maybe rename this to `onGamepadConnectionEvent`?

File: src/plugins/GamepadManager.js [hunk 2]
```diff
@@ -290,6 +284,22 @@ module.exports = class GamepadManager {
...
+     */
+    getRawGamepad(index) {
+        const rawGamepads = navigator.getGamepads();
+        for (let i = 0; i < rawGamepads.length; i++) {
+            if (rawGamepads[i].index === index) {
+                return rawGamepads[i];
+            }
+        }
+        // eslint-disable-next-line no-undefined
+        return undefined;
```
Comment (pgivel) [inline src/plugins/GamepadManager.js:300]: I think `return null` should be fine

File: src/plugins/GamepadManager.js [hunk 3]
```diff
@@ -351,7 +361,10 @@ module.exports = class GamepadManager {
      * @param {number} remoteIndex index of this gamepad in the remote VM
      */
     listenForInputs(localIndex, remoteIndex) {
```
Comment (pgivel) [inline src/plugins/GamepadManager.js:363]: I guess you could rename these to `hostIndex` and `guestIndex` for clarity & consistency with the desktop app

**Comment by cmajed** [comment]
yes it cames from this PR
https://github.com/Genymobile/genymotion-device-web-player/pull/72
I could target it instead!

File: src/plugins/GamepadManager.js [hunk 4]
```diff
@@ -351,7 +361,10 @@ module.exports = class GamepadManager {
...
-        this.currentGamepads[localIndex] = {remoteIndex: remoteIndex, buttons: []};
+        const gamepad = this.getRawGamepad(remoteIndex);
```
Comment (pgivel) [inline src/plugins/GamepadManager.js:364]: it's very weird to me that we use the remoteIndex here?  I think in the raw gamepads the `index` is the local one?

**Review by pgivel — CHANGES_REQUESTED** [review]

File: src/plugins/GamepadManager.js [hunk 5]
```diff
@@ -262,24 +262,26 @@ module.exports = class GamepadManager {
...
+            log.debug('GamepadManager unknown gamepad connection event');
         }
-        window.dispatchEvent(customEvent);
     }
 
     /**
      * Handler for gamepad disconnection. Also stops listening for inputs for this gamepad and emits a custom event
      * @param {GamepadEvent} event raw event coming from the browser Gamepad API
      */
     onGamepadDisonnected(event) {
```
Comment (pgivel) [inline src/plugins/GamepadManager.js:277]: oops I left a typo here
```suggestion
    onGamepadDisconnected(event) {
```

File: src/plugins/GamepadManager.js [hunk 6]
```diff
@@ -262,24 +262,26 @@ module.exports = class GamepadManager {
      * @param {GamepadEvent} event raw event coming from the browser Gamepad API
      */
     onGamepadConnected(event) {
-        const parsedGamepad = this.parseGamepad(event.gamepad);
-        const customEvent = new CustomEvent('gm-gamepadConnected', {detail: parsedGamepad});
-        if (event.gamepad.mapping === '') {
-            log.error(`Unsupported gamepad mapping for gamepad ${parsedGamepad.name}`);
-            return;
+        if (event.type === 'gamepadconnected') {
```
Comment (pgivel) [inline src/plugins/GamepadManager.js:265]: same here, I don't think this check is needed?

**Review by pgivel — CHANGES_REQUESTED** [review]

**Review by pgivel — APPROVED** [review]

**Pull request merged**